crates/parcel_core/src/plugin.rs

```diff
@@ -48,3 +48,12 @@ pub struct PluginOptions {
   pub mode: BuildMode,
   pub project_root: PathBuf,
 }
+
+impl From<&ParcelOptions> for PluginOptions {
+  fn from(options: &ParcelOptions) -> Self {
+    PluginOptions {
+      mode: options.mode.clone(),
```

Comment:
Can you just inline this

Reply:
There will be a conversion later when more of the options are ported over. The `PluginOptions` is a subset of `ParcelOptions` effectively as you can see in the JavaScript implementation [here](https://github.com/parcel-bundler/parcel/blob/v2/packages/core/core/src/public/PluginOptions.js) and the `From` will be a convenience when we get there.

Reply:
I'd rather use a constructor, because it seems you're constructing an invalid options state (without the root) then immediately patching it. Also it's used once.

If there's a full conversion then we can refactor to trait implementation.

We should always start by writing raw methods and functions.

From docs above

> The conversion is value-preserving: the conceptual kind and meaning of the resulting value is the same, even though the Rust type and technical representation might be different
> The conversion is obvious: it’s the only reasonable conversion between the two types. Otherwise it’s better to have it be a named method or constructor

Reply:
Yeah, I'm not fussed tbh and ended up inlining it